include/envoy/http/protocol.h

```diff
@@ -9,8 +9,8 @@ namespace Http {
  * Possible HTTP connection/request protocols. The parallel NumProtocols constant allows defining
  * fixed arrays for each protocol, but does not pollute the enum.
  */
-enum class Protocol { Http10, Http11, Http2 };
-const size_t NumProtocols = 3;
+enum class Protocol { Http10, Http11, Http2, Http3 };
```

Comment:
I think it's fine to update quiche/codec_impl.h and quiche_filter_manager and remove the TODO in this PR.

Reply:
I think anyone using these fields without them being plugged in is in for a world of trouble anyway, so I'm fine with it.  Might help as Lizan suggests to hide the option.

Reply:
SGTM. What's the preferred way to plug in QUIC codec into envoy client_codec and hcm_config then? Considering that envoy core code shouldn't depend on QUICHE, I could think of:
1)  a codec factory 
2) adding Network::Connection::createCodec() interface which is implemented to create a quic codec in quic connection and no-op in ConnectionImpl.

Any other ideas?

Reply:
What's ppl's thoughts on this?

Reply:
okay, I will plug in QUIC in the integration PR to make sure it's tested. And meanwhile I can think about which approach to take.